pkg/envoy/ads/stream.go

```diff
@@ -48,7 +49,8 @@ func (s *Server) StreamAggregatedResources(server xds_discovery.AggregatedDiscov
 	// This is the Envoy proxy that just connected to the control plane.
 	// NOTE: This is step 1 of the registration. At this point we do not yet have context on the Pod.
 	//       Details on which Pod this Envoy is fronting will arrive via xDS in the NODE_ID string.
-	proxy := envoy.NewProxy(kind, uuid, si, utils.GetIPFromContext(server.Context()))
+	streamID := atomic.AddInt64(&s.streamCount, 1)
+	proxy := envoy.NewProxy(kind, uuid, si, utils.GetIPFromContext(server.Context()), streamID)
```

Comment:
Looks like uuid is replaced by connectionID? We can get rid of it in future PR if that's easier.

Reply:
A good way to think about it is that a UUID maps to a proxy, while a ConnectionID maps to the connection (grpc Stream). If the same proxy connects and reconnects we need to know which stream, so we can remove the old one, and which UUID (to identify which proxy it is)